### ISSUE TITLE
Model loading and editing is slow in Alpha release code snapshot

### ISSUE THREAD

**Issue by aymanhab** [opensim-org/opensim-core#1472]
Building release version with Java wrapping on a beefed up machine results in test failure/timeout in test case Java_TestEditProperties_Release

Progress message in visual studio indicates
 Model loading and initialization time (ms):3501

The test case tries to perform 10 simple edits of a simple "double" property within 30 secs.

**Comment by aymanhab** [comment]
Tracked this down to Mesh::extendFinalizeFromProperties which gets called for every Mesh to load Geometry files during initSystem. The  edit sequence invokes initSystem twice (on a model and a clone). Restoring the old behavior where Geometry is loaded on demand only would solve this but maybe inconsistent with the workflow for Components. Any thoughts @aseth1 @chrisdembia @tkuchida

**Comment by tkuchida** [comment]
> The edit sequence invokes initSystem twice (on a model and a clone).

Why is it necessary to clone a model when performing a property edit?

**Comment by aymanhab** [comment]
Because an edit may result in an invalid model, so the code keeps a clone in order to recover

**Comment by tkuchida** [comment]
> Because an edit may result in an invalid model, so the code keeps a clone in order to recover

Does cloning automatically trigger a call to `initSystem()`? If so, perhaps we need an initSystem-less clone method since it sounds like the clone's call to `initSystem()` would be necessary only if the property edit results in an invalid model.

**Comment by aymanhab** [comment]
Nope, the cloning doesn't initSystem by itself. The call is done explicitly afterwards. Even one call to initSystem would still be mult-second operation since it unnecessarily requires parsing/loading of mesh files as it stands.

**Comment by tkuchida** [comment]
Why isn't it possible to do something like the following:
```cpp
auto& clonedModel = model.clone();
model.performPropertyEdit();
if (!model.valid()) {
    model = clonedModel;
    model.initSystem();
}
```

**Comment by aymanhab** [comment]
Basically editing consists of two stages. 

1. Change a property value and call initSystem (so you have a system consistent with the updated model/properties)
2. Copying state from pre-editing-model to post-editing-model so that values of state variables (e.g. configuration) are not lost.

The call sequence invoked since 3.0 or earlier is:
OpenSimContext::cacheModelAndState
.. Code to change Property value
OpenSimContext::restoreStateFromCachedModel.

It still feels that loading mesh files on initSystem is a bit excessive when they're used only for visualization. If I'm scripting or simulating I shouldn't care about these anyway AFAIK.

**Comment by aseth1** [comment]
The problem is not `initSystem`. The issue, I believe, is an error in the logic of `Mesh::extendFinalizeFromProperties()` which is supposed to cache the mesh to avoid reloading. I see it caches a `DecorativeMeshFile` which isn't caching the actual mesh and relies on `DecorativeMeshFile` to load the file. This seems rather pointless if the `SimTK::PolygonalMesh pmesh`  is already being tested for and loaded.

**Comment by aseth1** [comment]
It looks like `DecorativeMeshFile` is designed to do the loading upon visualization so it seems all `Mesh::extendFinalizeFromProperties()` need do is verify the existence of the file and not load it. The alternative is to cache the mesh itself. Right now the solution in place is mangling two potentially valid approaches into something that doesn't work.

**Comment by aymanhab** [comment]
Good point @aseth1 :+1: The code is trying to load the mesh file on initialization to avoid reporting the error due to bad files on each frame but that could be avoided some other way. I'll remove the loading and test with a bad/missing mesh file. Thank you. While initSystem still looks for all mesh files, it will not load them so that may be manageable.

**Comment by aymanhab** [comment]
Should be fixed by PR #1484

**Comment by aymanhab** [comment]
PR merged. Closing
